core/src/test/java/org/hibernate/ogm/test/batchfetching/BatchFetchingTest.java

```diff
@@ -0,0 +1,115 @@
+/*
+ * Hibernate OGM, Domain model persistence for NoSQL datastores
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.ogm.test.batchfetching;
+
+import org.fest.assertions.Assertions;
+
+import org.hibernate.Hibernate;
+import org.hibernate.Session;
+import org.hibernate.ogm.util.impl.Log;
+import org.hibernate.ogm.util.impl.LoggerFactory;
+import org.junit.Test;
+
+import org.hibernate.ogm.utils.OgmTestCase;
+import org.hibernate.stat.Statistics;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+/**
+ * @author Emmanuel Bernard emmanuel@hibernate.org
+ */
+public class BatchFetchingTest extends OgmTestCase {
+	@Override
+	protected Class<?>[] getAnnotatedClasses() {
+		return new Class<?>[] { Tower.class, Floor.class};
+	}
+
+	@Test
+	public void testLoadSeveralFloorsByBatch() throws Exception {
+		Session session = openSession();
+		Tower tower = prepareDataset( session );
+		session.clear();
+
+		Log log = LoggerFactory.make();
+		log.error( "Done with the insertion" );
+
+		session.beginTransaction();
+		for ( Floor currentFloor : tower.getFloors() ) {
+			// load proxies
+			assertFalse( Hibernate.isInitialized( session.load( Floor.class, currentFloor.getId() ) ) );
+		}
+		Statistics statistics = session.getSessionFactory().getStatistics();
+		statistics.setStatisticsEnabled( true );
+		statistics.clear();
+
+		assertEquals( 0, statistics.getEntityStatistics( Floor.class.getName() ).getFetchCount() );
+		for ( Floor currentFloor : tower.getFloors() ) {
+			// load proxies
+			Object entity = session.load( Floor.class, currentFloor.getId() );
+			Hibernate.initialize( entity );
+			assertTrue( Hibernate.isInitialized( entity ) );
+		}
+		assertEquals( 1, statistics.getEntityStatistics( Floor.class.getName() ).getFetchCount() );
```

Comment:
Apparently, still `getTuple()` is used underneath atm. How comes it still reports 1 as fetch count. I am wondering whether this test set-up is based on a wrong assumption?